src/OhmmsSoA/VectorSoaContainer.h

```diff
@@ -139,6 +140,15 @@ struct VectorSoaContainer
     if (nAllocated)
       myAlloc.deallocate(myData, nAllocated);
     nLocal     = n;
+    // Bail out to avoid some allocator implementations returning allocated ptr that
+    // must be freed, breaking the nAllocated scheme.
+    if(n == 0)
+    {
+      nAllocated = 0;
+      nGhosts = 0;
+      myData = nullptr;
+      return;
+    }
```

Comment:
I feel better with if-else. The code logic is more clear. return in the middle of the code is sub-optimal.

Reply:
Agree will clean up.

Reply:
restructured this code.